pkg/config/config.go

```diff
@@ -62,3 +62,11 @@ type Loader interface {
 	// Load the config.Provider
 	Load() (Provider, error)
 }
+
+// NameResolver interface to customize property recovery by name
+type NameResolver interface {
```

Comment:
Name is misleading in my opinion, you're actually resolving property values, not names...
```suggestion
type ValueResolver interface {
```